mantidproject/mantid
# Show Detector ROIs as Ranges on the Experiment Settings Tab

**Description of work.**
Make the Apply button on the preview tab fill the Experiment Settings tab's ROI Detector IDs column with ranges rather than an unreadable list of values. 

**To test:**
1. Open the ISIS Reflectometry interface
2. Go to the Preview tab
3. Enter run `INTER45455_inst` and click Load (This data is available on this Epic and must be in your path: #31069)
4. Select the `Draw a Rectangle` button on the instrument viewer. 
5. Draw over some of the detectors. 
6. Hit Apply
7. Check that the selected detectors are now present on the Experiment Settings tab's final column. 

Fixes #34606 


*This does not require release notes* because **this piece of work is not user facing yet.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
